### ISSUE TITLE
Other Roll-Up which isn't working

### ISSUE THREAD

**Issue by HarryRSamuel** [afawcett/declarative-lookup-rollup-summaries#391]
I have a field on the Account object called "Current Orders Completed" which is supposed to look at any Orders with Statuses "Processing" or "Completed" placed on that Account during the current report period. There's a field on Orders called "In Current Report Period" which is a checkbox formula that will be ticked if the Order Start Date is during the current report period. The Roll-Up is configured as follows:

![image](https://cloud.githubusercontent.com/assets/20912161/18019051/3322e1de-6bd2-11e6-94a6-8d705eb6275a.png)

Relationship Criteria are:

(In_Current_Reporting_Period__c = TRUE and (Status = 'PROCESSING' or Status = 'COMPLETED'))

The Account record I'm using to test is this:

![image](https://cloud.githubusercontent.com/assets/20912161/18019105/818efe16-6bd2-11e6-82b2-cb1ce8931f2b.png)

The related Order is this - I've highlighted the two criteria fields:

![image](https://cloud.githubusercontent.com/assets/20912161/18019128/9ee1dc68-6bd2-11e6-8909-0a52105acdf6.png)

I'd really appreciate any help you can offer!

**Comment by afawcett** [comment]
The tool will monitor for changes in fields on the child record that change during record updates on the child. In this case Status only. Your formula field 'In Current Report Period' will not work, because it cannot see changes in of formula fields (platform limitation). Depending on the formula the work around is to list the fields referenced int he formula also in the Relationship Criteria Fields field, e.g. your Order Start Date. So in realtime mode, any change to Status or Order Start Date (if specified on your rollup) should trigger a rollup for sure.

However since your using a time based formula "is during the current report period", this blows this workaround out of the water. Since your not reliant solely on child record insert, updates or deletes to trigger the rollup calc. You are also dependent on the passage of time! Even Schedule Mode = Scheduled has this issue (which still monitors the record, just differs updates on parent records to the background).

For time based rollup criteria you need to switch to using Schedule Calculate button to schedule a full recalc (you can provide criteria if needed). For this you don't need the child trigger deployed either. You can set the Calculation Mode to something like Process Builder (this is a bit of a hack to avoid the requirements of the other modes, i really need to create a new mode to reflect this use case). 

The above is assuming your checkbox formula is using a time based element in its expression, if you think i have made an invalid assumption let me know what your formula looks like and i'll have a re-think. Cheers!

**Comment by HarryRSamuel** [comment]
I've actually realised that the main reason this wasn't working is I gave the Relationship field the wrong name! I put Account instead of AccountID. Strangely when I created and saved the Roll-Up it didn't throw out an error which is why I didn't notice it, but when I tried to edit the test Order record today it gave me an error saying the "Account" field didn't exist.

I'm not sure that the time-based thing is going to be an issue for us since we will have orders and their statuses being created & updated on such a regular basis, so even if there is a bit of a time lag it won't be that significant. However we could schedule updates every hour (or even every minute) to reduce the time lag to a minimum. Are there any disadvantages to doing this in terms of using up platform limits etc?
